utils/sampler/lib/new_sample.dart

```diff
@@ -0,0 +1,114 @@
+
+import 'model.dart';
+
+typedef _OptionDialogSubmitted = void Function(Type sampleType, String? template);
+
+/// A modal dialog that allows selection of parameters for the new sample.
+class OptionDialog extends StatefulWidget {
+  const OptionDialog({Key? key, required this.onSubmitted}) : super(key: key);
+
+  final _OptionDialogSubmitted onSubmitted;
+
+  @override
+  _OptionDialogState createState() => _OptionDialogState();
+}
+
+class _OptionDialogState extends State<OptionDialog> {
+  String? selectedSampleType;
+  String? selectedTemplate;
+
+  @override
+  Widget build(BuildContext context) {
+    return AlertDialog(
+      title: const Text('Select Sample Type'),
+      content: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: <Widget>[
+          Row(
+            children: <Widget>[
+              const Text('Sample Type:'),
+              const Spacer(),
+              Padding(
+                padding: const EdgeInsets.all(8.0),
+                child: DropdownButton<String>(
+                  value: selectedSampleType,
+                  onChanged: (String? value) {
+                    setState(() {
+                      selectedSampleType = value;
+                    });
+                  },
+                  items: const <DropdownMenuItem<String>>[
+                    DropdownMenuItem<String>(child: Text('dartpad'), value: 'dartpad'),
+                    DropdownMenuItem<String>(child: Text('sample'), value: 'sample'),
+                    DropdownMenuItem<String>(child: Text('snippet'), value: 'snippet'),
+                  ],
+                ),
+              ),
+            ],
+          ),
+          if (selectedSampleType == 'sample' || selectedSampleType == 'dartpad')
+            Row(
+              children: <Widget>[
+                const Text('Template:'),
+                const Spacer(),
+                Padding(
+                  padding: const EdgeInsets.all(8.0),
+                  child: DropdownButton<String>(
+                    value: selectedTemplate,
+                    onChanged: (String? value) {
+                      setState(() {
+                        selectedTemplate = value;
+                      });
+                    },
+                    items: Model.instance
+                        .getTemplateNames()
+                        .map<DropdownMenuItem<String>>((String name) {
+                      return DropdownMenuItem<String>(child: Text(name), value: name);
+                    }).toList(),
+                  ),
+                ),
+              ],
+            ),
+        ],
+      ),
+      actions: <Widget>[
+        TextButton(
+          child: const Text('CANCEL'),
+          onPressed: () {
+            Navigator.of(context).pop();
+          },
+        ),
+        TextButton(
+          child: const Text('OK'),
+          onPressed: selectedSampleType != null &&
+                  (selectedTemplate != null || selectedSampleType == 'snippet')
```

Comment:
Formatting here is a little confusing. Perhaps have all of the conditions on a single line?

Reply:
OK, I rearranged it.